vault/resource_quota_lease_count_test.go

```diff
@@ -0,0 +1,89 @@
+package vault
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+	"github.com/hashicorp/vault/api"
+)
+
+func randomQuotaLeaseString() string {
+	whole := float64(acctest.RandIntRange(1000, 2000))
+	decimal := float64(acctest.RandIntRange(0, 100)) / 100
+
+	leaseCout := fmt.Sprintf("%.1f", whole+decimal)
+	// Vault retuns floats with trailing zeros trimmed
+	return strings.TrimRight(strings.TrimRight(leaseCout, "0"), ".")
+}
+
+func TestQuotaLeaseCount(t *testing.T) {
+	name := acctest.RandomWithPrefix("tf-test")
```

Comment:
Since this is an enterprise-only feature, I think this test will need to check for TF_ACC_ENTERPRISE at the top:

```suggestion
func TestQuotaLeaseCount(t *testing.T) {
	isEnterprise := os.Getenv("TF_ACC_ENTERPRISE")
	if isEnterprise == "" {
		t.Skip("TF_ACC_ENTERPRISE is not set, test is applicable only for Enterprise version of Vault")
	}

	name := acctest.RandomWithPrefix("tf-test")
```

Reply:
Updated